packages/ember-metal/lib/watch_key.js

```diff
@@ -28,10 +27,10 @@ export function watchKey(obj, keyName, meta) {
     }
 
     if (isEnabled('mandatory-setter')) {
-      handleMandatorySetter(m, obj, keyName);
+      handleMandatorySetter(meta, obj, keyName);
```

Comment:
depending on https://github.com/emberjs/ember.js/pull/12852/files#r50556996 we may be able to skip this if desc is already present.